src/egl/drivers/dri2/platform_android.c

```diff
@@ -222,6 +224,28 @@ droid_window_dequeue_buffer(struct dri2_egl_surface *dri2_surf)
    return EGL_TRUE;
 }
 
+static int
+droid_resolve_format(struct dri2_egl_display *dri2_dpy,
+                     struct ANativeWindowBuffer *buf)
+{
+   int format;
+   int err;
+
+   if (buf->format != HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED)
+      return buf->format;
+
+   if (!dri2_dpy->gralloc->perform)
+      return -1;
+
+   err = dri2_dpy->gralloc->perform(dri2_dpy->gralloc,
```

Comment:
Looks like Gralloc 1.0's perform call has support for the GRALLOC_DRM_GET_FORMAT operation. So I don't think any change is required, still, I will test it and report back.